src/world/Actor/Chara.cpp

```diff
@@ -536,6 +535,18 @@ void Chara::statusEffectFreeSlot( uint8_t slotId )
   m_statusEffectFreeSlotQueue.push( slotId );
 }
 
+void Chara::replaceSingleStatusEffectById( uint32_t id )
+{
+  for( const auto& effectIt : m_statusEffectMap )
+  {
+    if( effectIt.second->getId() == id )
+    {
+      removeStatusEffect( effectIt.first, false );
+      break;
+    }
+  }
+}
```

Comment:
This is a map, why do you iterate it to find an id? You should be able to jund "find()" it.